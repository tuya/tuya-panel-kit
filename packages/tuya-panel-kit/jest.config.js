const base = require('../../jest.config.base.js');
const pkg = require('./package');

module.exports = {
  ...base,
  name: pkg.name,
  displayName: pkg.name,
  rootDir: '../..',
  coverageDirectory: '<rootDir>/packages/tuya-panel-kit/coverage/',
  testMatch: ['<rootDir>/packages/tuya-panel-kit/src/components/**/__tests__/?(*.)+(test).js'],
  collectCoverageFrom: [
    '<rootDir>/packages/tuya-panel-kit/src/components/**/*.js',
    '<rootDir>/packages/tuya-panel-kit/src/utils/*.js',
    '<rootDir>/packages/tuya-panel-kit/src/utils/theme/*.js',
    '!<rootDir>/packages/tuya-panel-kit/src/components/layout/+(navigator-layout|full-view|offline-view|api|detect-net-modal|react-navigation)/*.js',
    '!<rootDir>/packages/tuya-panel-kit/src/components/layout/offline-view/ble-offline-view/*.js',
    '!<rootDir>/packages/tuya-panel-kit/src/components/dialog/password.js',
    '!<rootDir>/packages/tuya-panel-kit/src/components/popup/dropdown/*.js',
    '!<rootDir>/packages/tuya-panel-kit/src/components/feature-slider/*.js',
    '!<rootDir>/packages/tuya-panel-kit/src/index.js',
    '!<rootDir>/packages/tuya-panel-kit/src/helpers/*.js',
    '!<rootDir>/packages/tuya-panel-kit/src/components/barchart/*.js',
    '!<rootDir>/packages/tuya-panel-kit/src/components/linechart/*.js',
    '!<rootDir>/packages/tuya-panel-kit/src/components/image-upload/*.js',
    '!<rootDir>/packages/tuya-panel-kit/src/components/i18n/*.js',
    '!<rootDir>/packages/tuya-panel-kit/src/components/theme/*.js',
    '!<rootDir>/packages/tuya-panel-kit/src/TYNativeApi.js',
  ],
  modulePathIgnorePatterns: [
    '<rootDir>/example/node_modules',
    '<rootDir>/demos',
    '<rootDir>/example',
    '<rootDir>/lib',
    '<rootDir>/packages/*/lib',
    '<rootDir>/packages/tuya-panel-utils',
  ],
};
