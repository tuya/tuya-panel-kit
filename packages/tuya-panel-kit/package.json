{
  "name": "tuya-panel-kit",
  "version": "4.10.0",
  "description": "a functional component library for developing tuya device panels!",
  "keywords": [
    "tuya",
    "react-native",
    "panel"
  ],
  "author": "ZengBeauty<2497752603@qq.com>",
  "homepage": "https://github.com/tuya/tuya-panel-kit",
  "license": "MIT",
  "main": "src/index.js",
  "directories": {
    "lib": "lib",
    "test": "__tests__"
  },
  "files": [
    "lib",
    "src"
  ],
  "publishConfig": {
    "registry": "https://registry.npmjs.org"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/tuya/tuya-panel-kit.git"
  },
  "scripts": {
    "prebuild": "rimraf lib/*",
    "build": "babel src --ignore __tests__ --out-dir lib --copy-files",
    "dev": "babel src --watch --ignore __tests__ --out-dir lib --copy-files",
    "test": "NODE_ENV=testing TZ=UTC jest -u --silent --forceExit",
    "test:watch": "jest --watch",
    "prepublishOnly": "yarn build"
  },
  "bugs": {
    "url": "https://github.com/tuya/tuya-panel-kit/issues"
  },
  "dependencies": {
    "@react-native-community/masked-view": "0.1.10",
    "@react-navigation/native": "5.7.3",
    "@react-navigation/stack": "5.9.0",
    "babel-cli": "^6.26.0",
    "color": "^0.11.1",
    "events": "^1.1.1",
    "hoist-non-react-statics": "^3.3.0",
    "prop-types": "^15.6.1",
    "react-addons-shallow-compare": "^15.6.2",
    "react-native-deprecated-custom-components": "0.1.2",
    "react-native-gesture-handler": "1.3.0",
    "react-native-reanimated": "1.9.0",
    "react-native-safe-area-context": "3.0.7",
    "react-native-screens": "2.9.0",
    "react-native-svg": "5.5.1",
    "style-equal": "^1.0.0",
    "styled-components": "^3.4.10",
    "tuya-panel-animation-diffusion": "^0.2.1",
    "tuya-panel-animation-drawer": "^0.2.1",
    "tuya-panel-animation-wave": "^0.2.1",
    "tuya-panel-icon": "^0.1.3",
    "tuya-panel-linear-gradient": "^0.1.1",
    "tuya-panel-slider": "^0.1.1",
    "tuya-panel-stepper": "^0.1.3",
    "tuya-panel-switch": "^0.1.5",
    "tuya-panel-tab": "^0.1.1",
    "tuya-panel-tabbar": "^0.1.2",
    "tuya-panel-tabs": "^0.1.3",
    "tuya-panel-theme": "^0.1.2",
    "tuya-panel-utils": "^0.2.1"
  },
  "peerDependencies": {
    "@types/tuya-panel-kit": "*",
    "react": "^16.8.3",
    "react-native": "^0.59.10",
    "react-native-svg": "^5.5.1"
  }
}
