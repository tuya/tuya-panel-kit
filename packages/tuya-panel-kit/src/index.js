import * as Utils from './utils';

// 主题配置Provider组件  + 默认主题配置变量.
// export { Theme, defaultTheme } from './components/theme';

// The useTheme hook let's us access the currently active theme.
export { useTheme, Theme, defaultTheme } from 'tuya-panel-theme';

export { default as createNavigator } from './components/layout/react-navigation';

export { default as TransitionPresets } from './components/layout/react-navigation/TransitionPresets';

export { default as FullView } from './components/layout/full-view';

export { default as NavigatorLayout } from './components/layout/navigator-layout';

export { default as OfflineView } from './components/layout/offline-view';

export { default as TopBar } from './components/layout/topbar';

export { default as CircleView } from './components/circle-view';

export { default as Checkbox } from './components/checkbox';

export { default as Slider } from './components/slider';

export { default as SwitchButton } from './components/switch-button';

export { default as BrickButton } from './components/button-brick';

export { default as Button } from './components/button';

export { default as Divider } from './components/divider';

export { default as LinearGradient } from './components/gradient/linear-gradient';

export { default as Motion } from './components/motion';

export { default as RadialGradient } from './components/gradient/radial-gradient';

export { default as TYSectionList } from './components/TYLists/lists';

export { default as TYFlatList } from './components/TYLists/list';

export { default as Stepper } from './components/stepper';

export { default as TYListItem } from './components/TYLists/list-item';

export { default as Toast } from './components/toast-view';

export { default as Tips } from './components/tips';

export { default as RotationView } from './components/rotation-view';

export { default as IconFont } from './components/iconfont';

export { default as UnitText } from './components/unit-text';

export { default as TYText } from './components/TYText';

export { default as Picker } from './components/picker-view';

export { default as DatePicker } from './components/date-picker';

export { default as Swipeout } from './components/swipeout';

export { default as I18N } from './components/i18n';

export { Strings, TYSdk } from './TYNativeApi';

export { Utils };

export { default as Modal } from './components/modal';

export { default as Collapsible } from './components/collapsible';

export { default as Tab } from './components/tab';

export { default as Tabs } from './components/tabs';

export { default as TabBar } from './components/tabbar';

export { default as Carousel } from './components/carousel';

export { default as Dialog } from './components/dialog';

export { default as Popup } from './components/popup';

export { default as ControllerBar } from './components/controller-bar';

export { default as Notification } from './components/notification';

export { default as NotificationLegacy } from './components/notification-legacy';

export { default as TimerPicker } from './components/timer-picker';

export { default as Progress } from './components/progress';

export { default as GlobalToast } from './components/global-toast';

export { default as Battery } from './components/battery';

export { default as SliderProgress } from './components/slider-progress';

export { default as SliderWithLine } from './components/slider-with-line';

export { default as TopBarMallIndicator } from 'tuya-panel-topbar-mall-indicator';
