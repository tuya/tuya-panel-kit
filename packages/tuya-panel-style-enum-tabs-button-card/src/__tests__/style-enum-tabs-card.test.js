/**
 * @jest-environment jsdom
 */

import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import { ClassicEnumTabsButtonCard } from '../index'
import TuyaRNSvgs from 'tuya-panel-kit/lib/components/iconfont/svg/defaultSvg';
import { TabBar } from 'tuya-panel-kit'

const IMAGE =
  'data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAADMAAAAzCAYAAAA6oTAqAAAAEXRFWHRTb2Z0d2FyZQBwbmdjcnVzaEB1SfMAAABQSURBVGje7dSxCQBACARB+2/ab8BEeQNhFi6WSYzYLYudDQYGBgYGBgYGBgYGBgYGBgZmcvDqYGBgmhivGQYGBgYGBgYGBgYGBgYGBgbmQw+P/eMrC5UTVAAAAABJRU5ErkJggg==';

const data = [
  {
    label: '1',
    key: '1',
    disabled: true,
  },
  {
    label: '2',
    key: 'two',
  },
  {
    label: '3',
    key: '3',
  },
  {
    label: '4',
    key: '5',
  },
  {
    label: '5',
    key: '6',
  },
  {
    label: '6',
    key: '7',
  },
];

describe('EnumTabsCard', () => {
  it ('Classic', () => {
    const wrap = renderer.create(
      <ClassicEnumTabsButtonCard
        data={data}
        title="Classic"
        icon={TuyaRNSvgs.power}
        unit="档"
      />
    ).toJSON();

    expect(wrap).toMatchSnapshot();  
  });

  it('Classic set props', () => {
    const wrap = renderer.create(
      <ClassicEnumTabsButtonCard
        data={data}
        title="Classic"
        icon={IMAGE}
        iconIsImage={true}
        unit="档"
      />
    ).toJSON();

    expect(wrap).toMatchSnapshot(); 
  })

  it('in controll', () => {
    let activeKey = '1';
    const changeActiveKey = (key) => activeKey = key;
    const wrap = mount(
      <ClassicEnumTabsButtonCard
        data={data}
        title="Classic"
        icon={TuyaRNSvgs.power}
        unit="档"
        activeKey={activeKey}
        onChange={changeActiveKey}
      />
    );
    const tabInstance = wrap.find(TabBar).at(0);
    tabInstance.props().tabs[2].onItemPress();
    wrap.update();
    expect(wrap.find(TabBar).at(0).props().activeKey).toBe('1');
    expect(activeKey).toBe('3');
  });

  it('not controll', () => {
    const wrap = mount(
      <ClassicEnumTabsButtonCard
        data={data}
        title="Classic"
        defaultActiveKey="1"
      />
    );
    const tabInstance = wrap.find(TabBar).at(0);
    tabInstance.props().tabs[2].onItemPress();
    wrap.update();
    expect(wrap.find(TabBar).at(0).props().activeKey).toBe('3');
  })

  it('disabled', () => {
    const wrap = mount(
      <ClassicEnumTabsButtonCard
        data={data}
        title="Classic"
        icon={TuyaRNSvgs.power}
        unit="档"
        defaultActiveKey="3"
      />
    );
    const tabInstance = wrap.find(TabBar).at(0);
    tabInstance.props().tabs[0].onItemPress();
    wrap.update();
    expect(wrap.find(TabBar).at(0).props().activeKey).toBe('3');
  })
});

