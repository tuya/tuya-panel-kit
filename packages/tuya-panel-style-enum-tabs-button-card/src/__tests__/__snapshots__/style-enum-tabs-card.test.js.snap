// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EnumTabsCard Classic 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
        "borderRadius": 28,
        "paddingBottom": 44,
        "paddingLeft": 40,
        "paddingRight": 40,
        "paddingTop": 44,
      },
      Object {},
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        },
        Object {
          "marginBottom": 48,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "flexDirection": "row",
          },
          Object {
            "height": 28,
            "width": 28,
          },
          false,
          null,
        ]
      }
    >
      <RNSVGSvgView
        align="xMidYMid"
        meetOrSlice={0}
        minX={0}
        minY={0}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
            },
            undefined,
            false,
            Object {
              "flex": 0,
              "height": 28,
              "width": 28,
            },
          ]
        }
        vbHeight={1024}
        vbWidth={1024}
      >
        <RNSVGPath
          d="M512 0C794.760533 0 1024 229.239467 1024 512S794.760533 1024 512 1024 0 794.760533 0 512 229.239467 0 512 0z m185.048178 327.0656a26.988089 26.988089 0 0 0-38.183822 38.183822 207.712711 207.712711 0 1 1-293.728712 0 26.988089 26.988089 0 1 0-38.183822-38.183822c-102.172444 102.1952-102.172444 267.901156 0 370.096356 102.1952 102.172444 267.901156 102.172444 370.096356 0 102.172444-102.1952 102.172444-267.901156 0-370.096356zM511.886222 227.555556a27.079111 27.079111 0 0 0-26.919822 24.302933l-0.136533 2.776178v196.152889a27.079111 27.079111 0 0 0 53.998933 2.776177l0.136533-2.776177v-196.152889a27.079111 27.079111 0 0 0-27.079111-27.079111z"
          fill={
            Array [
              0,
              0.23921568627450981,
              0.23921568627450981,
              0.23921568627450981,
              1,
            ]
          }
          fillOpacity={1}
          fillRule={1}
          matrix={
            Array [
              1,
              0,
              0,
              1,
              0,
              0,
            ]
          }
          opacity={1}
          propList={
            Array [
              "stroke",
              "strokeWidth",
              "fill",
            ]
          }
          stroke={
            Array [
              0,
              0.23921568627450981,
              0.23921568627450981,
              0.23921568627450981,
              1,
            ]
          }
          strokeDasharray={null}
          strokeDashoffset={null}
          strokeLinecap={0}
          strokeLinejoin={0}
          strokeMiterlimit={4}
          strokeOpacity={1}
          strokeWidth={1}
        />
      </RNSVGSvgView>
    </View>
    <Text
      accessible={true}
      align={null}
      allowFontScaling={false}
      color={null}
      ellipsizeMode="tail"
      size={30}
      style={
        Array [
          false,
          Array [
            false,
            Array [
              Object {
                "backgroundColor": "transparent",
                "color": "#333",
                "fontSize": 30,
              },
              Array [
                Object {
                  "color": "#3d3d3d",
                  "fontSize": 30,
                  "fontWeight": "normal",
                  "marginLeft": 16,
                },
                Object {},
              ],
            ],
          ],
        ]
      }
      type={null}
      weight={null}
    >
      Classic
    </Text>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "marginLeft": 20,
        }
      }
    >
      <Text
        accessible={true}
        align={null}
        allowFontScaling={false}
        color={null}
        ellipsizeMode="tail"
        size={30}
        style={
          Array [
            false,
            Array [
              false,
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "color": "#333",
                  "fontSize": 30,
                },
                Object {
                  "color": "#158CFB",
                  "fontSize": 30,
                  "fontWeight": "normal",
                  "marginRight": 10,
                },
              ],
            ],
          ]
        }
        type={null}
        weight={null}
      >
        ·
      </Text>
      <Text
        accessible={true}
        align={null}
        allowFontScaling={false}
        color={null}
        ellipsizeMode="tail"
        size={30}
        style={
          Array [
            false,
            Array [
              false,
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "color": "#333",
                  "fontSize": 30,
                },
                Object {
                  "color": "#158CFB",
                  "fontSize": 30,
                  "fontWeight": "normal",
                  "marginRight": 10,
                },
              ],
            ],
          ]
        }
        type={null}
        weight={null}
      >
        1
        档
      </Text>
    </View>
  </View>
  <View
    style={Object {}}
  >
    <View
      onLayout={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "#E3E9EE",
            "borderColor": "#fff",
            "borderRadius": 30,
            "height": 60,
            "justifyContent": "center",
          },
          Object {
            "backgroundColor": "#E3E9EE",
          },
          Object {
            "paddingHorizontal": 16,
          },
        ]
      }
    >
      <View
        onLayout={[Function]}
        style={
          Array [
            Object {
              "flexDirection": "row",
            },
            Object {},
          ]
        }
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "justifyContent": "center",
              "opacity": 1,
              "width": 0,
            }
          }
        >
          <Text
            accessible={true}
            align={null}
            allowFontScaling={false}
            color={null}
            ellipsizeMode="tail"
            numberOfLines={1}
            size={28}
            style={
              Array [
                false,
                Array [
                  false,
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                      "color": "#333",
                      "fontSize": 28,
                    },
                    Array [
                      Object {
                        "backgroundColor": "transparent",
                        "color": "#9CA3A9",
                        "fontSize": 14,
                        "fontWeight": "400",
                        "textAlign": "center",
                      },
                      Array [
                        Array [
                          Object {
                            "opacity": 0,
                          },
                        ],
                        Object {},
                        undefined,
                      ],
                      Object {
                        "color": "#57BCFB",
                        "fontWeight": "500",
                      },
                      Object {
                        "color": "#fff",
                        "fontSize": 28,
                      },
                    ],
                  ],
                ],
              ]
            }
            type={null}
            weight={null}
          >
            1
          </Text>
        </View>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "justifyContent": "center",
              "opacity": 1,
              "width": 0,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "#57BCFB",
                  "borderRadius": 2,
                  "height": 4,
                  "width": 4,
                },
                Object {
                  "backgroundColor": "#158CFB",
                  "borderRadius": 8,
                  "height": 8,
                  "opacity": 0.4,
                  "width": 8,
                },
              ]
            }
          />
        </View>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "justifyContent": "center",
              "opacity": 1,
              "width": 0,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "#57BCFB",
                  "borderRadius": 2,
                  "height": 4,
                  "width": 4,
                },
                Object {
                  "backgroundColor": "#158CFB",
                  "borderRadius": 8,
                  "height": 8,
                  "opacity": 0.4,
                  "width": 8,
                },
              ]
            }
          />
        </View>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "justifyContent": "center",
              "opacity": 1,
              "width": 0,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "#57BCFB",
                  "borderRadius": 2,
                  "height": 4,
                  "width": 4,
                },
                Object {
                  "backgroundColor": "#158CFB",
                  "borderRadius": 8,
                  "height": 8,
                  "opacity": 0.4,
                  "width": 8,
                },
              ]
            }
          />
        </View>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "justifyContent": "center",
              "opacity": 1,
              "width": 0,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "#57BCFB",
                  "borderRadius": 2,
                  "height": 4,
                  "width": 4,
                },
                Object {
                  "backgroundColor": "#158CFB",
                  "borderRadius": 8,
                  "height": 8,
                  "opacity": 0.4,
                  "width": 8,
                },
              ]
            }
          />
        </View>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "justifyContent": "center",
              "opacity": 1,
              "width": 0,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "#57BCFB",
                  "borderRadius": 2,
                  "height": 4,
                  "width": 4,
                },
                Object {
                  "backgroundColor": "#158CFB",
                  "borderRadius": 8,
                  "height": 8,
                  "opacity": 0.4,
                  "width": 8,
                },
              ]
            }
          />
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`EnumTabsCard Classic set props 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
        "borderRadius": 28,
        "paddingBottom": 44,
        "paddingLeft": 40,
        "paddingRight": 40,
        "paddingTop": 44,
      },
      Object {},
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        },
        Object {
          "marginBottom": 48,
        },
      ]
    }
  >
    <Image
      source={
        Object {
          "uri": "data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAADMAAAAzCAYAAAA6oTAqAAAAEXRFWHRTb2Z0d2FyZQBwbmdjcnVzaEB1SfMAAABQSURBVGje7dSxCQBACARB+2/ab8BEeQNhFi6WSYzYLYudDQYGBgYGBgYGBgYGBgYGBgZmcvDqYGBgmhivGQYGBgYGBgYGBgYGBgYGBgbmQw+P/eMrC5UTVAAAAABJRU5ErkJggg==",
        }
      }
      style={
        Object {
          "borderRadius": 0,
          "height": 28,
          "width": 28,
        }
      }
    />
    <Text
      accessible={true}
      align={null}
      allowFontScaling={false}
      color={null}
      ellipsizeMode="tail"
      size={30}
      style={
        Array [
          false,
          Array [
            false,
            Array [
              Object {
                "backgroundColor": "transparent",
                "color": "#333",
                "fontSize": 30,
              },
              Array [
                Object {
                  "color": "#3d3d3d",
                  "fontSize": 30,
                  "fontWeight": "normal",
                  "marginLeft": 16,
                },
                Object {},
              ],
            ],
          ],
        ]
      }
      type={null}
      weight={null}
    >
      Classic
    </Text>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "marginLeft": 20,
        }
      }
    >
      <Text
        accessible={true}
        align={null}
        allowFontScaling={false}
        color={null}
        ellipsizeMode="tail"
        size={30}
        style={
          Array [
            false,
            Array [
              false,
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "color": "#333",
                  "fontSize": 30,
                },
                Object {
                  "color": "#158CFB",
                  "fontSize": 30,
                  "fontWeight": "normal",
                  "marginRight": 10,
                },
              ],
            ],
          ]
        }
        type={null}
        weight={null}
      >
        ·
      </Text>
      <Text
        accessible={true}
        align={null}
        allowFontScaling={false}
        color={null}
        ellipsizeMode="tail"
        size={30}
        style={
          Array [
            false,
            Array [
              false,
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "color": "#333",
                  "fontSize": 30,
                },
                Object {
                  "color": "#158CFB",
                  "fontSize": 30,
                  "fontWeight": "normal",
                  "marginRight": 10,
                },
              ],
            ],
          ]
        }
        type={null}
        weight={null}
      >
        1
        档
      </Text>
    </View>
  </View>
  <View
    style={Object {}}
  >
    <View
      onLayout={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "#E3E9EE",
            "borderColor": "#fff",
            "borderRadius": 30,
            "height": 60,
            "justifyContent": "center",
          },
          Object {
            "backgroundColor": "#E3E9EE",
          },
          Object {
            "paddingHorizontal": 16,
          },
        ]
      }
    >
      <View
        onLayout={[Function]}
        style={
          Array [
            Object {
              "flexDirection": "row",
            },
            Object {},
          ]
        }
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "justifyContent": "center",
              "opacity": 1,
              "width": 0,
            }
          }
        >
          <Text
            accessible={true}
            align={null}
            allowFontScaling={false}
            color={null}
            ellipsizeMode="tail"
            numberOfLines={1}
            size={28}
            style={
              Array [
                false,
                Array [
                  false,
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                      "color": "#333",
                      "fontSize": 28,
                    },
                    Array [
                      Object {
                        "backgroundColor": "transparent",
                        "color": "#9CA3A9",
                        "fontSize": 14,
                        "fontWeight": "400",
                        "textAlign": "center",
                      },
                      Array [
                        Array [
                          Object {
                            "opacity": 0,
                          },
                        ],
                        Object {},
                        undefined,
                      ],
                      Object {
                        "color": "#57BCFB",
                        "fontWeight": "500",
                      },
                      Object {
                        "color": "#fff",
                        "fontSize": 28,
                      },
                    ],
                  ],
                ],
              ]
            }
            type={null}
            weight={null}
          >
            1
          </Text>
        </View>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "justifyContent": "center",
              "opacity": 1,
              "width": 0,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "#57BCFB",
                  "borderRadius": 2,
                  "height": 4,
                  "width": 4,
                },
                Object {
                  "backgroundColor": "#158CFB",
                  "borderRadius": 8,
                  "height": 8,
                  "opacity": 0.4,
                  "width": 8,
                },
              ]
            }
          />
        </View>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "justifyContent": "center",
              "opacity": 1,
              "width": 0,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "#57BCFB",
                  "borderRadius": 2,
                  "height": 4,
                  "width": 4,
                },
                Object {
                  "backgroundColor": "#158CFB",
                  "borderRadius": 8,
                  "height": 8,
                  "opacity": 0.4,
                  "width": 8,
                },
              ]
            }
          />
        </View>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "justifyContent": "center",
              "opacity": 1,
              "width": 0,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "#57BCFB",
                  "borderRadius": 2,
                  "height": 4,
                  "width": 4,
                },
                Object {
                  "backgroundColor": "#158CFB",
                  "borderRadius": 8,
                  "height": 8,
                  "opacity": 0.4,
                  "width": 8,
                },
              ]
            }
          />
        </View>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "justifyContent": "center",
              "opacity": 1,
              "width": 0,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "#57BCFB",
                  "borderRadius": 2,
                  "height": 4,
                  "width": 4,
                },
                Object {
                  "backgroundColor": "#158CFB",
                  "borderRadius": 8,
                  "height": 8,
                  "opacity": 0.4,
                  "width": 8,
                },
              ]
            }
          />
        </View>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "justifyContent": "center",
              "opacity": 1,
              "width": 0,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "#57BCFB",
                  "borderRadius": 2,
                  "height": 4,
                  "width": 4,
                },
                Object {
                  "backgroundColor": "#158CFB",
                  "borderRadius": 8,
                  "height": 8,
                  "opacity": 0.4,
                  "width": 8,
                },
              ]
            }
          />
        </View>
      </View>
    </View>
  </View>
</View>
`;
